pkg/upgradehelper/versionguard/version.go

```diff
@@ -11,11 +11,21 @@ import (
 
 func Check(upgrade *v1beta1.Upgrade, strictMode bool, minUpgradableVersionStr string) error {
 
-	currentVersion, err := version.NewHarvesterVersion(upgrade.Status.PreviousVersion)
+	repoInfo, err := getRepoInfo(upgrade)
```

Comment:
Not related to this PR, but I'm curious why UpgradeStatus.RepoInfo type is string
https://github.com/harvester/harvester/blob/848f704bbdbd577305b5500d1978bddd784b16b4/pkg/apis/harvesterhci.io/v1beta1/upgrade.go#L54

instead of type `RepoInfo`, is there any concern ?

https://github.com/harvester/harvester/blob/848f704bbdbd577305b5500d1978bddd784b16b4/pkg/controller/master/upgrade/repoinfo/repoinfo.go#L15-L17

Reply:
Thanks for the explanation ! I'm not sure if we can convert the type from string to RepoInfo, will try to do that if have time. If that's something we can improve, I'll create an issue later.